anydex/test/wallet/stellar/test_stellar_wallet.py

```diff
@@ -0,0 +1,308 @@
+        """
+        Test for get_transactions when wallet has not been created
+        """
+        self.assertEqual([], await self.wallet.get_transactions())
+
+    async def test_get_transactions_not_created(self):
+        """
+        Test for get_transactions when wallet has not been created
+        """
+        self.assertEqual([], await self.wallet.get_transactions())
+
+    async def test_get_transactions_created(self):
+        """
+        Test for get_transactions when wallet has been created
+        """
+        # using the global tx doesn't work for some reason
+        tx = Transaction(hash='96ad71731b1b46fceb0f1c32adbcc16a93cefad1e6eb167efe8a8c8e4e0cbb98',
+                         ledger_nr=26529414,
+                         date_time=datetime.fromisoformat('2020-06-05T08:45:33'),
+                         source_account='GDQWI6FKB72DPOJE4CGYCFQZKRPQQIOYXRMZ5KEVGXMG6UUTGJMBCASH',
+                         operation_count=1,
+                         transaction_envelope="AAAAAOFkeKoP9De5JOCNgRYZVF8IIdi8WZ6olTXYb1KTMlgRAAAAZAGOO"
+                                              "+wAAAA9AAAAAQAAAAAAAAAAAAAAAAAAAAAAAAADAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                                              "AAAAAAAAAAAAAAABAAAAAQAAAADhZHiqD/Q3uSTgjYEWGVRfCCHYvFmeqJU12G9SkzJYEQA"
+                                              "AAAAAAAAAXQbflbQZVvhfaHtF6ESvgGrNnl2gi44084MWUaGbmNkAAAAAAcnDgAAAAAAAAA"
+                                              "ABHvBc2AAAAEBUL1wo8IGHEpgpQ7llGaFE+rC9v5kk2KPJe53/gIdWF+792HYg5yTTmhJII"
+                                              "97YgM+Be8yponPH0YjMjeYphewI",
+                         fee=100,
+                         is_pending=False,
+                         succeeded=True,
+                         sequence_number=112092925529161789,
+                         min_time_bound=datetime.fromisoformat('1970-01-01T00:00:00'))
+        self.wallet.stellar_db.insert_transaction(tx)
+        self.wallet.get_address = lambda: 'GBOQNX4VWQMVN6C7NB5UL2CEV6AGVTM6LWQIXDRU6OBRMUNBTOMNSOAW'
+        self.wallet.provider = MockObject()
+        self.wallet.provider.get_ledger_height = lambda: 26529414
+        self.wallet.provider.get_transactions = lambda *_: []
+        self.wallet.created = True
+        payment_dict = {
+            'id': '96ad71731b1b46fceb0f1c32adbcc16a93cefad1e6eb167efe8a8c8e4e0cbb98',  # use tx hash for now
+            'outgoing': False,
+            'from': 'GDQWI6FKB72DPOJE4CGYCFQZKRPQQIOYXRMZ5KEVGXMG6UUTGJMBCASH',
+            'to': 'GBOQNX4VWQMVN6C7NB5UL2CEV6AGVTM6LWQIXDRU6OBRMUNBTOMNSOAW',
+            'amount': 30000000,
+            'fee_amount': 100,
+            'currency': self.expected_identifier(),
+            'timestamp': 1591339533.0,
+            'description': f'confirmations: 1'
+        }
+
+        self.assertEqual([payment_dict], await self.wallet.get_transactions())
+
+    @timeout(10)
+    async def test_monitor_transactions_found(self):
```

Comment:
Test fails with `concurrent.futures._base.TimeoutError` for concrete wallet class, not testnet class.